### ISSUE TITLE
4.8 Java

### ISSUE THREAD

**Issue by birdyhuang** [careercup/ctci#146]
The question said both trees are very large. Does it mean we should not use a recursive way of doing this?

**Comment by piyush121** [comment]
Complexity will remain same in case of recursion.

**Comment by birdyhuang** [comment]
But recursion may get stack overflow, doesn't it?

**Comment by piyush121** [comment]
Yeah it will. But I think we have to assume that our system can handle O(n) space.

**Comment by nitinpunjabi** [comment]
For what it's worth, I didn't use recursion for my solution precisely because it mentioned the trees are large.

**Comment by birdyhuang** [comment]
Then why does the question mention the size of the tree? Is it just useless or some kinda clue?
